packages/webapp/src/utils/api.tsx

```diff
@@ -464,9 +464,16 @@ export function useGetProvidersAPI() {
 export function useGetConnectionListAPI() {
     const signout = useSignout();
 
-    return async () => {
+    return async (limit: number, offset: number, integration?: string) => {
         try {
-            let res = await fetch('/api/v1/connection', { headers: getHeaders() });
+            const res = await fetch(
+                `/api/v1/connection?limit=${limit}&offset=${offset}` +
+                    `${integration ? `&integration=${integration}` : ''}`,
```

Comment:
can we use `new URLSearchParams({ foo: 'value' })` instead of building the url manually